### DISCUSSION TITLE
URL scheme must be  "http" or "https" for CORS request

### DISCUSSION THREAD

**Discussion by zifahm** [vercel/next.js#14219] [Help]
This happens when I use with-styletron
Im also using base web UI.

```
worker.js:1 Fetch API cannot load webpack-internal:///./node_modules/baseui/esm/layer/layers-manager.js. URL scheme must be "http" or "https" for CORS request
```

Any Idea how to solve this issue?

**Comment by thiagodebastos** [comment]
@zifahm did you find a solution? I'm facing the same issue.

**Reply by jk7g14** [reply]
same problem here using nextjs 9.5.3

**Comment by amponsah** [comment]
This error is with with-styletron example setup... Any ideas on how to fix this?

**Reply by millievn** [reply]
i meet this issue now. any ideas here?

**Comment by JiveCheng** [comment]
if you use webpack in development env, you can setting devtool: 'inline-cheap-source-map', it will be solved

**Reply by devalnor** [reply]
It work in v9.4.4 but v9.5 seems to break this solution
More info #14906

**Comment by candidia** [comment]
I too get this in Next js using styletron and base web. I used the example here https://github.com/vercel/next.js/tree/canary/examples/with-styletron

**Comment by maciekChmura** [comment]
I'm using Blitz and Base Web and also have this issue.
```
Fetch API cannot load webpack-internal:///../../node_modules/baseui/esm/layer/layers-manager.js. URL scheme "webpack-internal" is not supported.
```

**Comment by simplenotezy** [comment]
Getting same error. It happened out of nowhere. Not sure why.

```
injectGlobalHook.js:1648 Fetch API cannot load webpack-internal:///./pages/projects/[projectId]/users/[[...userId]].js. URL scheme must be "http" or "https" for CORS request.

injectGlobalHook.js:1648 Fetch API cannot load webpack-internal:///./node_modules/swr/esm/use-swr.js. URL scheme must be "http" or "https" for CORS request.
```

**Reply by CoteTommy** [reply]
Same thing for me

**Reply by magisters-cc** [reply]
Same. Just throw errors in dev mode. In production mode console is clean.
![image](https://user-images.githubusercontent.com/12679280/132808454-306701d4-8528-4e73-ab7c-78a7dda7d1a2.png)

**Reply by magisters-cc** [reply]
For my case I found out the reason. This happens after Chrome was updated. At the same time i use previous version of Chrome Canary, and the errors disappeared.

**Reply by mrv1k** [reply]
**Upd2:** 
Issue reappeared. Again, toggled it off/on, disappeared ¯\_(ツ)_/¯

**Upd1:** 
I think I fixed it by accident. The issue was caused by **React Developer Tools**.
Not sure what exactly was the problem but, _ahem_
1. Turned extension off, the issue disappeared
2. Turned extension back on expecting for the issue to reappear. It didn't. Now can't reproduce.

Can somebody please confirm?

Idk, stale cache or something? I didn't even have to reinstall Dev tools. Just toggle off and on.
Pretty sure the reason it didn't appear in Brave is because it was a fresh browser install and clean Dev Tools.

---
**Original post* 
---
@magisters-cc is right. Chrome only issue. Appears with dynamic imports

Appears in Chrome: `Version 93.0.4577.82 (Official Build) (x86_64)`
Doesn't in Brave: `Version 1.29.81 Chromium: 93.0.4577.82 (Official Build) (x86_64)`

Repro:
```jsx
/* Ok.jsx */
export default function Ok() {
  return <div>ok</div>;
}
```
```jsx
/* Ok.page.jsx */
import dynamic from "next/dynamic";

const Ok = dynamic(() => import("./Ok.jsx"));

export default function OkPage() {
  return (
    <div>
      <Ok />
    </div>
  );
}
```

**Reply by simplenotezy** [reply]
Can confirm it has something to do with React Dev Tools

**Reply by sedghi** [reply]
I'm actually getting this error when i was debugging my react app with the "component" tab in the react-devTools. I wanted to look into the named hooks option, and clicking on my components gives this error

**Comment by AbdullahDiaa** [comment]
I'm getting same error:
`Fetch API cannot load webpack-internal:///./node_modules/@headlessui/react/dist/components/transitions/transition.esm.js. URL scheme "webpack-internal" is not supported.
onMessage @ injectGlobalHook.js:1648
postMessage (async)
(anonymous) @ VM33451:2`

**Comment by gino** [comment]
Having the same issue, not quite sure why this is happening:

![image](https://user-images.githubusercontent.com/23482089/133432510-fd07cab0-8bb4-4ae5-ad8f-635799be1fb0.png)

**Comment by maciekChmura** [comment]
With Blitz and Base Web I "fixed it" for now by removing the `debug={debug} debugAfterHydration` from `StyletronProvider`
```js
<StyletronProvider value={styletron} debug={debug} debugAfterHydration>
<StyletronProvider value={styletron}> 
```

**Comment by samora17** [comment]
Getting same error. Custom hook of my component: 
<code> useMyCustomHook = ({ componentRenderingData }) => { ...
useEffect(() => {
    resetComponentToInitState();
  }, [componentRenderingData]);

  const resetComponentToInitState = () => {
    dispatch({
      type: TYPES.COMPONENT_UPDATE_DATA,
      payload: initComponent(componentRenderingData),
    });
  };</code>

ERROR : 
<code> injectGlobalHook.js:1648 Fetch API cannot load webpack-internal:///./components/myComponent.js. URL scheme "webpack-internal" is not supported.

onMessage  @injectGlobalHook.js:1648
postMessage (async)
(anonymous) @VM1834:2 </code>

<img width="878" alt="Captura de Pantalla 2021-09-17" src="https://user-images.githubusercontent.com/86170988/133853993-08928c7e-b5d8-454b-b26c-3b6eb36dbac2.png">

**Comment by proaleks94** [comment]
same there
![image](https://user-images.githubusercontent.com/29139456/135027851-128f8cf1-7778-44b7-a58e-83c42b4f124f.png)

**Comment by jacobdubail** [comment]
I'm running Next 11.1.2
React Query 3.21
Graphql Request 3.5.0

Seeing the same error in my console "URL scheme "webpack-internal" is not supported."

**Comment by NatanCieplinski** [comment]
The issue is probably React Dev Tools extension. Try turning it off and then back on and the error should disappear.

**Reply by fabcodingzest** [reply]
Hey, it did disappear but when you go to the components tab, it makes the appearance again :/
Do I need to worry about this ? or will it go away in production?

**Reply by NatanCieplinski** [reply]
I'm not sure but i don't think so since the issue is only related to the devtool extension

**Reply by nyxarr** [reply]
Hi, it has been resolved in last version of Chrome 94.0.4606.71

**Reply by ivanvanderbyl** [reply]
I've isolated this issue to the setting `Always parse hook names from source (may be slow)` in React Developer Tools. When this is turned on the issue is present.

Edit: Using Chrome 96.0.4664.93

**Reply by falcon11** [reply]
Yes, I meet the same problem with the setting. Hoping to be resolved.
